File: plugins/vbnet-kit/testdata/bad/smell.vb
' Intentional smells for vbnet-rg-gate discrimination (not product code).
Option Strict Off

Public Module Smell
    Public Sub BadPath()
        On Error Resume Next
        Console.WriteLine("lib console leak")
        Dim x = "1"
        Dim y As Integer = x
    End Sub
End Module

File: plugins/delphi-kit/testdata/bad/smell.pas
unit Smell;

{ Intentional smells for delphi-rg-gate discrimination (not product code). }

interface

implementation

type
  TThing = record
    X: Integer;
    Y: Integer;
  end;

procedure BadPath;
label
  Done;
var
  P: Pointer;
  T: TThing;
begin
  GetMem(P, 64);
  with T do
  begin
    X := 1;
    Y := 2;
  end;
  WriteLn('lib console leak');
  goto Done;
Done:
end;

end.
